docs/getting-started/overview.rst

```diff
@@ -45,35 +47,27 @@ All operations take the variances into account:
   The implemented mechanism assumes uncorrelated data.
 
 
-Event data
-----------
+Scattered data (event data)
```

Comment:
Should we use scattered, when the main driver for the development of scipp is neutron scattering? In that sense both event and dense data are scattered. Maybe sparse instead?

Reply:
The problem with "sparse" is that the term is widely used and understood as in https://en.wikipedia.org/wiki/Sparse_matrix. There are numerous "sparse (matrix) data formats" that store such matrices/arrays in a packed manner, for example there is https://docs.scipy.org/doc/scipy/reference/sparse.html.

So for me "sparse" feels like a no-go. I though "scattered" was a common and widely understood term? Would it b sufficient if we explained it in more detail?

Reply:
Explaining might be helpful, though given the loaded terms "scattered" and "sparse" maybe `non-dense` is a better descriptor.

Reply:
Why do you consider "scattered" loaded? Just because it is in "neutron scattering"?

I wold argue that it is more important to use standard terminology, see, e.g., https://www.google.com/search?q=scatter+data&client=firefox-b-d&source=lnms&tbm=isch&sa=X&ved=2ahUKEwix__enwaLtAhUai8MKHTKqDmEQ_AUoAXoECA4QAw&biw=1240&bih=703&dpr=2 and https://en.wikipedia.org/wiki/Scatter_plot.

You can even argue that this is the *same* meaning as in neutron scattering, after all neutrons are scattered to (and detected at) different points in space.

Reply:
What if you store scattering data in a dense format?

Is the event data layout really that different from sparse layouts?

Reply:
The "desne" format we use for storig neutron-scattering data is actually still "scattered": The histogrammed event data is typically still scattered in space, i.e., there are detector pixels at scattered points in space:

```
dense = Variable(dims=['position', 'tof'], ...) # scattered in space
events = Variable(dims=['position'], ...) # scattered in space and time (time is internal coord of the event list in every bin
```

And yes, the scattering data is stored in a dense format: In the case of binned data the internal buffer is a dense table. The "scatter data" aspects becomes apparent when we then use coord in that table to plot things in a scatter plot.

Reply:
That sounds like all data is 'scattered'. So is the term meaningless? How doe what you said relate to 'event data'?

Also, I am not aware of any sparse matrix layout that has significant gaps in memory. It is a ind of dense table which is just interpreted in a certain way. Don't we do the same thing?

Reply:
> Also, I am not aware of any sparse matrix layout that has significant gaps in memory. It is a ind of dense table which is just interpreted in a certain way. Don't we do the same thing?

Not really: Sparse matrix layouts are a space-efficient way of storing *regular* "grids" with many zeros. Scattered data does not fall on a grid (unless we artificially introduce one). For example event time-of flight may be measured in `nanosecond`, stored as single or double precision `int` of `float` (the NeXus standard is flexible), i.e., the data is completely irregular.

> That sounds like all data is 'scattered'. So is the term meaningless? How doe what you said relate to 'event data'?

No, not all data is scattered. Maybe we can say that all *raw* data *in neutron scattering* is scattered (either in space if data is histogrammed before writing to files, or in space and time if data is written as "event data"). However:
- A lot of the reduced data is dense, e.g., `I(Q)` or `I(Q_x, Q_y, Q_z)`.
- Other data such as monitors are just a dense 1D array, not scattered.
- Data from imaging sensors (or any other rectangular detectors) is not scattered.
- A lot of non-neutron application may have dense data.

All I am trying to say above is that there is a step wise transition between scattered data and dense data. In the case of neutron reduction:

1. event data recorded in irregularly spaced pixels (scattered in time and space).
2. histogram data recorded in irregularly spaced pixels (scattered in space).
3. dense data, e.g., if we would also histogram in space.

My point is simply that `event-data == (1.)` but `scattered-data == (1.) and (2.)`.

Reply:
Obviously we are discussing semantics here, so what I say is not 100% set in stone and usage of terminology differs. I suppose we just have to make up our minds, try to be as close to common usage as possible, and stick with it?